src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesReferenceService.kt

```diff
@@ -60,11 +88,38 @@ class IncentivesReferenceService(
     }
   }
 
+  suspend fun prisonIncentiveLevelChange(event: PrisonIncentiveChangedEvent) {
+    val incentiveLevelCode = event.additionalInformation.incentiveLevel
+    val prisonId = event.additionalInformation.prisonId
+    nomisApiService.getPrisonIncentiveLevel(prisonId, incentiveLevelCode)?.also {
+      incentivesApiService.getPrisonIncentiveLevel(prisonId, incentiveLevelCode)
+        .also { prisonIncentiveLevelData ->
+          nomisApiService.updatePrisonIncentiveLevel(prisonId, prisonIncentiveLevelData.toNomisPrisonIncentiveLevel())
+          trackEvent("prison-incentive-level-updated", prisonIncentiveLevelData)
+        }
+    } ?: also {
+      incentivesApiService.getPrisonIncentiveLevel(prisonId, incentiveLevelCode)
+        .also { prisonIncentiveLevelData ->
+          nomisApiService.createPrisonIncentiveLevel(prisonId, prisonIncentiveLevelData.toNomisPrisonIncentiveLevel())
+          trackEvent("prison-incentive-level-inserted", prisonIncentiveLevelData)
```

Comment:
Do we need some telemetry if the API calls fails for some reason?

Reply:
I don't think it adds much in this simpler process (no mapping calls). I might come back to it after testing on dev.